contracts/vesting.token/src/vesting.token.cpp

```diff
@@ -0,0 +1,72 @@
+#include <vesting.token/vesting.token.hpp>
+
+namespace vestingtoken {
+    void vestingToken::updatedate(eosio::time_point_sec newStartDate) {
+        require_auth(get_self());
+        startDate startDate(get_self(), get_self().value);
+        startDate.set(newStartDate, get_self());
+    }
+    void vestingToken::assigntokens(eosio::name holder, eosio::asset amount, VestingCategory category) {
+        // Only the contract owner can call this function
+        require_auth(get_self());
+
+        // Check the symbol is correct and valid
+        auto sym = amount.symbol;
+        eosio::check(sym == system_resource_currency, "Symbol does not match system resource currency");
+        eosio::check(sym.precision() == system_resource_currency.precision(), "Symbol precision does not match");
+        eosio::check(amount.amount > 0, "Amount must be greater than 0");
+
+        // Create a new vesting schedule
+        vesting_schedules vestingTable(get_self(), holder.value);
+        auto iter = vestingTable.end();
+
+        vestingTable.emplace(get_self(), [&](auto& row) {
+            row.holder = holder;
+            row.totalTokens = amount;
+            row.tokensClaimed = asset(0, amount.symbol);
+            row.allocated = eosio::current_time_point();
+            row.vestingCategory = category;
+        });
+    }
+    void vestingToken::withdraw(eosio::name holder){
+        require_auth(holder);
+        // Get the vesting schedule
+        vesting_schedules vestingTable(get_self(), holder.value);
+        for (auto iter = vestingTable.begin(); iter != vestingTable.end(); ++iter) {
+            const vested_allocation& vestingSchedule = *iter;
+
+            VestingCategory category = vestingSchedule.vestingCategory;
+
+            eosio::time_point_sec startDelayDays = eosio::time_point_sec(category.startDelayDays * 24 * 60 * 60);
+            eosio::time_point_sec startDateValue = startDate(get_self(), get_self().value).get();
+            eosio::time_point_sec vesting_start = startDateValue + eosio::seconds(startDelayDays.sec_since_epoch()) + vestingSchedule.allocated;
```

Comment:
eosio::seconds: This is a duration type that represents a period of time. It is used to represent a specific length of time, such as the duration of an event or the difference between two time points.

eosio::time_point_sec: This is a point in time, represented as the number of seconds since the Unix epoch (1970-01-01T00:00:00Z). It is used to represent a specific moment in time.